app/views/thredded/preferences/_form.html.erb

```diff
@@ -11,29 +11,43 @@
   <ul class="thredded--form-list">
     <li>
       <%= f.label :follow_topics_on_mention do %>
-        <%= f.check_box :follow_topics_on_mention %>
+        <%= f.check_box :follow_topics_on_mention, 'data-thredded-bound-messageboard-pref' => 'user_preferences_form[messageboard_follow_topics_on_mention]' %>
         <%= t 'thredded.preferences.form.follow_topics_on_mention.label' %>
         <p class="thredded--form-list--hint">
           <%= t 'thredded.preferences.form.follow_topics_on_mention.hint' %>
         </p>
       <% end %>
     </li>
     <li>
-      <%= f.label :followed_topic_emails do %>
-        <%= f.check_box :followed_topic_emails %>
-        <%= t 'thredded.preferences.form.followed_topic_emails.label' %>
-        <p class="thredded--form-list--hint">
-          <%= t 'thredded.preferences.form.followed_topic_emails.hint' %>
-        </p>
-      <% end %>
+      <label><%= t 'thredded.preferences.form.notifications_for_followed_topics.label' %></label>
+      <!--<p class="thredded&#45;&#45;form-list&#45;&#45;hint">-->
+        <!--<%= t 'thredded.preferences.form.notifications_for_followed_topics.hint' %>-->
+      <!--</p>-->
+      <%= f.fields_for :notifications_for_followed_topics do |fn| %>
+        <%- Thredded.notifiers.each do |notifier| %>
+          <%= fn.label notifier.key do %>
+            <%= fn.check_box notifier.key, checked: f.object.notifications_for_followed_topics[notifier.key] ?
```

Comment:
As I understand, this manual `checked` assignment, the `TRUTHY_HASH` loop in the controller, and the manual merging in the form model, all happen because `notifications_for_followed_topics` is a hash so it doesn't `respond_to?(notifier.key)`.

Perhaps a lot of this complexity could go away if `notifications_for_followed_topic` becomes a form object?